File: src/main/java/com/dh/ondot/member/core/OauthProviderConverter.java
package com.dh.ondot.member.core;

import com.dh.ondot.core.exception.UnsupportedException;
import org.springframework.core.convert.converter.Converter;
import org.springframework.stereotype.Component;
import com.dh.ondot.member.domain.enums.OauthProvider;

import static com.dh.ondot.core.exception.ErrorCode.UNSUPPORTED_SOCIAL_LOGIN;
import static java.util.Locale.ENGLISH;

@Component
public class OauthProviderConverter implements Converter<String, OauthProvider> {
    @Override
    public OauthProvider convert(String type) {
        try {
            return OauthProvider.valueOf(type.toUpperCase(ENGLISH));
        } catch (IllegalArgumentException e) {
            throw new UnsupportedException(UNSUPPORTED_SOCIAL_LOGIN, type);
        }
    }
}

File: src/main/kotlin/com/dh/ondot/member/core/OauthProviderConverter.kt
package com.dh.ondot.member.core

import com.dh.ondot.core.exception.ErrorCode.UNSUPPORTED_SOCIAL_LOGIN
import com.dh.ondot.core.exception.UnsupportedException
import com.dh.ondot.member.domain.enums.OauthProvider
import org.springframework.core.convert.converter.Converter
import org.springframework.stereotype.Component
import java.util.Locale

@Component
class OauthProviderConverter : Converter<String, OauthProvider> {
    override fun convert(type: String): OauthProvider =
        try {
            OauthProvider.valueOf(type.uppercase(Locale.ENGLISH))
        } catch (e: IllegalArgumentException) {
            throw UnsupportedException(UNSUPPORTED_SOCIAL_LOGIN, type)
        }
}
